src/main/java/appeng/util/inv/AdaptorItemHandler.java

```diff
@@ -0,0 +1,247 @@
+package appeng.util.inv;
+
+
+import java.util.Iterator;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+import appeng.api.config.FuzzyMode;
+import appeng.util.InventoryAdaptor;
+import appeng.util.Platform;
+
+
+public class AdaptorItemHandler extends InventoryAdaptor
+{
+	private final IItemHandler itemHandler;
+
+	public AdaptorItemHandler( IItemHandler itemHandler )
+	{
+		this.itemHandler = itemHandler;
+	}
+
+	@Override
+	public ItemStack removeItems( int amount, ItemStack filter, IInventoryDestination destination )
+	{
+		int slots = itemHandler.getSlots();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'slots' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)